runtimes/runtimes/standalone.ts

```diff
@@ -181,6 +182,61 @@ export const standalone = (props: RuntimeProps) => {
                         })
                     }),
                 getFileSize: path => stat(path),
+                getServerDataFolder: serverName => {
+                    const initializeParams = lspRouter.clientInitializeParams
+                    const clientSpecifiedLocation = initializeParams?.initializationOptions?.aws.clientDataFolder
+                    if (clientSpecifiedLocation) {
+                        return path.join(clientSpecifiedLocation, serverName)
+                    }
+
+                    const clientFolderName = getClientNameFromParams(initializeParams)
+                    const standardizedClientFolderName = standardizeFileName(clientFolderName)
+
+                    const appDataFolder = getPlatformAppDataFolder()
+                    return appDataFolder
+                        ? path.join(appDataFolder, standardizedClientFolderName, serverName)
+                        : path.join(os.homedir(), `.${standardizedClientFolderName}`, serverName)
+
+                    function getPlatformAppDataFolder(): string {
+                        const platform = process.platform
+                        switch (platform) {
+                            case 'win32':
+                                return process.env.APPDATA || path.join(os.homedir(), 'AppData', 'Roaming')
+
+                            case 'darwin':
+                                return path.join(os.homedir(), 'Library', 'Application Support')
+
+                            case 'linux':
+                                return process.env.XDG_DATA_HOME || path.join(os.homedir(), '.local', 'share')
+
+                            default:
+                                return ''
+                        }
+                    }
+
+                    function getClientNameFromParams(initializeParams: InitializeParams | undefined): string {
+                        const clientInfo = initializeParams?.clientInfo
+                        const awsClientInfo = initializeParams?.initializationOptions?.aws.clientInfo
+
+                        return [
+                            awsClientInfo?.name || clientInfo?.name || '',
+                            awsClientInfo?.version || clientInfo?.version || '',
+                            awsClientInfo?.extension.name || '',
+                            awsClientInfo?.extension.version || '',
+                        ]
+                            .filter(Boolean)
+                            .join('_')
+                    }
+
+                    function standardizeFileName(clientFolderName: string): string {
```

Comment:
nit: Maybe call this `standardizeDirName` given that if used on a file, it would convert the dot extension separator to an underscore.  While it's safe to use in this context right now, this function has the potential to be lifted to a global library and you wouldn't want to use it as-is against filenames.

Reply:
Agreed, I'll update it.